### PR TITLE
Handle RUN_CNI_WATCHDOG from all exit paths

### PR DESCRIPTION
**Pull request by jingyuanliang**

### CODE REVIEW

File: scripts/install-cni.sh [hunk 1]
```diff
@@ -50,6 +51,24 @@ echo "Install-CNI ($0), Build: $BUILD"
 
 set -u -e
 
+# Try to decouple RUN_CNI_WATCHDOG and ENABLE_CILIUM_PLUGIN; don't assume
+# ENABLE_CILIUM_PLUGIN is set whenever RUN_CNI_WATCHDOG is set.
+# All exit paths should call exit_success instead of `exit 0`,
+# so the appropriate actions can be taken according to RUN_CNI_WATCHDOG.
+exit_success() {
```
Comment (pravk03) [inline scripts/install-cni.sh:58]: rename ? 

We do not exit when watchdog is enabled

Comment (jingyuanliang) [reply scripts/install-cni.sh:58]: Do you have a recommendation? I'm not good at doing naming...

Comment (jingyuanliang) [reply scripts/install-cni.sh]: Done

File: scripts/install-cni.sh [hunk 2]
```diff
@@ -405,40 +419,30 @@ cilium_health_check() {
...
+# Cilium health check logic before watchdog and CNI STATUS API are introduced.
+cilium_wait_or_ignore() {
+  if cilium_health_check "${CILIUM_HEALTH_MAX_WAIT_TIME:-600}"; then
+    echo "Cilium healthz reported success."
+  else
+    echo "Cilium not yet ready. Continuing anyway."
   fi
+}
 
+write_and_success() {
```
Comment (pravk03) [inline scripts/install-cni.sh:431]: Maybe changing this to write_cni_conf and making a separate call to success is cleaner.

**Review by pravk03 — APPROVED** [review]

File: scripts/install-cni.sh (hunk 2 above)
Comment (jingyuanliang) [reply scripts/install-cni.sh:431]: It's already merged. Let's leave it as-is.

**Pull request merged**